### DISCUSSION TITLE
how to try erlang with unikraft ?

### DISCUSSION THREAD

**Discussion by 221V** [unikraft/unikraft#274] [Ideas]
[there are](https://github.com/unikraft/kraft) nice list:  
C, C++, Go, Python, Ruby, Lua, Javascript, WASM, Nginx, Redis, SQlite;

one question: how to try erlang with unikraft ?

thanks

**Comment by nderjung** [comment]
Hey! Thanks for opening this discussion.

After a little investigation, it looks like a port of [Erlang/OTP](https://github.com/erlang/otp) will be necessary to make erlang on Unikraft to work.  This has not yet been done, so it can created as a new project for Unikraft.

**Reply by arturovm** [reply]
Hello, @nderjung! What would that port entail? Do you have a general idea?

**Reply by nderjung** [reply]
Hi @arturovm,

To get started, try following this tutorial: https://usoc21.unikraft.org/docs/sessions/08-basic-app-porting/

This should cover the same ideas needed to get Erlang on Unikraft :) The tutorial covers building an application as a Unikernel, you can think of Erlang/OTP as an application which accepts an input program written in Erlang.  Follow the tutorial with the mindset that you're working with the source code for Erlang/OTP whose interpreter is written in C++.

**Comment by razvand** [comment]
@221V, since we are already (viciously) using [the Unikraft Discord server](https://bit.ly/UnikraftDiscord) for discussions, meetings, brainstorming and jokes (mostly bad, we do what we can), we will close the Discussions tab. We will do this on Friday, March 31, 2023.

Please move any discussions or questions on [the Unikraft Discord server](https://bit.ly/UnikraftDiscord). See you all there! :handshake:

**Comment by HoseaFranklin87** [comment]
Install Unikraft by following the instructions in the Unikraft documentation.
Download the Erlang source code and compile it using the Unikraft toolchain.
Write a Unikraft application that uses the Erlang runtime and build it using the Unikraft build system.
Run the Unikraft application on a virtual machine or a cloud platform that supports Unikraft.
Keep in mind that the process of using Erlang with Unikraft can be complex and may require advanced knowledge of both technologies.
